evaluators/jsonpath-evaluator/components/org.wso2.carbon.gateway.evaluators.jsonpathevaluator/src/main/java/org/wso2/carbon/gateway/evaluators/jsonpathevaluator/JSONPathEvaluator.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ * <p>
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.gateway.evaluators.jsonpathevaluator;
+
+import com.jayway.jsonpath.JsonPath;
+import org.wso2.carbon.gateway.core.Constants;
+import org.wso2.carbon.gateway.core.flow.contentaware.abstractcontext.MessageBodyEvaluator;
+import org.wso2.carbon.gateway.core.flow.contentaware.exceptions.MessageBodyEvaluationException;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * This class evaluates jsonpath expressions.
```

Comment:
Could you elaborate more on why this class exists in this world and what's the responsibility etc 
draw example from JDK class lib it self :) 
e.g. https://docs.oracle.com/javase/7/docs/api/java/util/LinkedList.html